src/core/api/option_utils.ts

```diff
@@ -414,6 +416,13 @@ function checkReloadOptions(options?: {
       options?.reloadAt?.relative !== undefined) {
     throw new Error("API: reload - Invalid 'reloadAt.relative' option format.");
   }
+  if (options?.keySystems === null ||
+      (!Array.isArray(options?.keySystems) && options?.keySystems !== undefined)) {
+    throw new Error("API: reload - Invalid 'keySystems' option format.");
+  }
```

Comment:
What is the point of checking if `options?.keySystems === null` ? 
If it's `null` it's not an Array nor `undefined` 
So I'm thinking the condition: 
`(!Array.isArray(options?.keySystems) && options?.keySystems !== undefined)` is enough

Reply:
You're right, even the undefined check should be unnecessary I guess

Reply:
No sorry I just re-read what I wrote, the undefined check is still necessary as that's a potential input but we also want to be tolerant and authorize `null`. But the conditions could be make better by checking `isNullOrUndefined`

Reply:
Wait no, we actually DIDN't authorize it being set to `null` apparently, you were right and I had a bug there :p.

Reply:
Thanks, I fixed it and push-forced to keep the same commit.

I removed the previous comments because that was just errors on my part